src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,628 @@
+        boolean fullMatch = low <= Integer.parseInt(barcodeString) && high >= Integer.parseInt(barcodeString);
+
+        // Biopsy Slides and RP Slides are a hybrid between SampleLabel and TimePoint: they have the same SampleLabel numbers and should be
+        // treated as sample labels in how they're displayed in tables, but they differ by their Times Received number (which is time point
+        // coding) so we need to check them differently here
+        if (this == BIOPSY_SLIDES) {
+          return fullMatch && TimePoint.BIOPSY.predicate().test(sample);
+        } else if (this == RP_SLIDES) {
+          return fullMatch && TimePoint.RADICAL.predicate().test(sample);
+        } else {
+          return fullMatch;
+        }
+      };
+    }
+  }
+
+  private enum Site {
+    // Coded by the first four digits of the donor number: PRE_*####*111_Xx_X_nn_1-1
+    SUNNYBROOK("Sunnybrook (1039)", "1039"), UHN("UHN (1042)", "1042"), LONDON("London (1023)", "1023"), //
+    MONTREAL("Montreal (2024)", "2024"), VANCOUVER("Vancouver (2026)", "2026");
+    private final String key;
+    private final String siteId;
+    private static final Map<String, Site> lookup = new TreeMap<>();
+    static {
+      for (Site s : EnumSet.allOf(Site.class)) {
+        lookup.put(s.getKey(), s);
+      }
+    }
+
+    Site(String key, String siteId) {
+      this.key = key;
+      this.siteId = siteId;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    public Predicate<SampleDto> predicate() {
+      return sample -> sample.getName().startsWith("PRE_" + siteId);
+    }
+  }
+
+  private enum TimePoint {
+    // Coded by the Times Received portion of the tissue name: PRE_1111111_Xx_X_nn_*#*-1
+    RANDOMIZATION("Screening/Randomization", 1), SIX("6 Month Follow Up", 2), TWELVE("12 Month Follow Up", 3), //
+    EIGHTEEN("18 Month Follow Up", 4), TWENTY_FOUR("24 Month Follow Up", 5), BIOPSY("Biopsy", 6), RADICAL("Radical Prostatectomy", 7);
+    private final String key;
+    private final int timePointCode;
+    private static final Map<String, TimePoint> lookup = new TreeMap<>();
+    static {
+      for (TimePoint s : EnumSet.allOf(TimePoint.class)) {
+        lookup.put(s.getKey(), s);
+      }
+    }
+
+    TimePoint(String key, int timePointCode) {
+      this.key = key;
+      this.timePointCode = timePointCode;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    /**
+     * Assumes that PRECISE samples will always have _nn_ in the name, and that Tissue Origin will never be "nn".
+     * Should hopefully be a safe assumption since this is a biobanking project where tracking tissue origin is critical.
+     */
+    public Predicate<SampleDto> predicate() { 
+      return sample -> {
+        if (SAMPLE_CATEGORY_IDENTITY.equals(sample.getSampleType())) return false;
+        return timePointCode == getTimesReceived(sample.getName());
+      };
+    }
+  };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)